lib/Core/ExecutionState.cpp

```diff
@@ -34,6 +34,9 @@ using namespace klee;
 namespace { 
   cl::opt<bool>
   DebugLogStateMerge("debug-log-state-merge");
+
+  // HACK: This is really not good for library design
+  uint64_t globalExecutionStateCounter = 0;
```

Comment:
I wasn't trying to add thread safety, it is not enough of course. What I think is a good tradeoff is to wrap it in a functions, so that if thread safety can be added quite easily.